base/reduce.jl

```diff
@@ -353,6 +358,43 @@ function extrema(itr)
     return (vmin, vmax)
 end
 
+"""
+    extrema(A,dims) -> Array{Tuple}
+
+Compute the minimum and maximum elements of an array over the given dimensions.
+"""
+function extrema(A::AbstractArray, dims)
+    sz = [size(A)...]
+    sz[[dims...]] = 1
+    B = Array{Tuple{eltype(A),eltype(A)}}(sz...)
+    extrema!(B, A)
+end
+
+@generated function extrema!{T,N}(B, A::Array{T,N})
+    quote
+        sA = size(A)
+        sB = size(B)
+        @nloops $N i B begin
+            AI = @nref $N A i
+            (@nref $N B i) = (AI, AI)
+        end
+        Bmax = sB
+        Istart = ones(Int,ndims(A))
+        Istart[([sB...].==1) & ([sA...].!=1)] = 2
+        @inbounds @nloops $N i d->(Istart[d]:size(A,d)) begin
+            AI = @nref $N A i
+            @nexprs $N d->(j_d = min(Bmax[d], i_{d}))
+            BJ = @nref $N B j
+            if AI < BJ[1]
+              (@nref $N B j) = (AI, BJ[2])
+            elseif AI > BJ[2]
+              (@nref $N B j) = (BJ[1], AI)
+            end
+        end
+        squeeze(B, tuple(find([sB...].==1)...))
```

Comment:
I'd get rid of this squeeze for consistency with `minimum(A, dims)`, `maximum(A, dims)`, etc.